### DISCUSSION TITLE
Feedback Wanted: LLM Prompts/Evaluation Workflow in MLflow

### DISCUSSION THREAD

**Discussion by adogra-db** [mlflow/mlflow#8822] [General]
Hi everyone- I'm a PM working on MLflow. We are actively building better support for LLM development workflows (for prompt engineering and evaluation) within MLflow. 

**If you have had any experience tuning and evaluating LLMs** (whether on MLflow or elsewhere) to build any kind of application (e.g. chatbot, etc) and would be willing to share some feedback that'll impact how we best support you, **please reply to this discussion and/or email me at archika.dogra@databricks.com.** Thank you!

**Comment by cdecoux** [comment]
This feedback is specific to utilizing MLFlow on Databricks with GPU Model Serving and Dolly V2:

I've had quite a bit of issues following some basic examples online. I've been utilizing the LangChain flavor but could never seem to get it working on Databricks correctly.

One example I followed was this one: https://www.dbdemos.ai/minisite/llm-dolly-chatbot/# (via Live Demo sometime in May). While the LangChain models worked directly, logging them via Databrick's MLFlow Model Registry either was unsuccessful or broken. The closest I've gotten to get it working on Databricks was updating all the versions of LangChain and MLFlow and utilizing LangChain's HuggingFaceHub API to pull the LLM into a LLMChain. This requires API keys that unfortunately can't seem to be passed as environment variables to the model.

Somewhere in the PRs or Issues here on https://github.com/mlflow, I saw a comment about how LangChain produces breaking changes pretty often. Looking at https://github.com/mlflow/mlflow/blob/master/requirements/extra-ml-requirements.txt seems to show that `langchain` does not have a pinned version, so it's not really clear which version of LangChain will be working for me or if I have a potentially breaking version.

**Reply by cdecoux** [reply]
Coming back to this, I switched to the HuggingFace flavor for MLFlow and had a way better experience with this. For my case, this could potentially still be leveraged in LangChain with its [Databricks integration](https://python.langchain.com/docs/modules/model_io/models/llms/integrations/databricks).

I think the issues with LangChain were restrictions on models that rely on remote code (like Dolly and MosaicML MPT models), but it was definitely difficult to debug on MLFlow.

**Comment by shimizust** [comment]
Hi, Our team at LinkedIn is evaluating MLflow for prompt engineering and LLM tracking. We're interested in understanding the roadmap for this particular aspect of the product. Given that @adogra-db might have transitioned from Databricks, could you kindly direct us to the right contact or team to discuss this in more detail?

Thanks!

**Reply by AbeOmor** [reply]
Hi @shimizust, I'm a PM working on PM MLflow in Databricks. Please reach out at abe.omorogbe@databricks.com, would love to learn how you are using LLMs with MLflow

**Reply by kjoth** [reply]
@shimizust Whats does you mean by  LLM tracking using Flow. We are exploring options of MLFlow, not for LLM hsoting, but can MLFlow can be used as Observability & Monitroring for LLM APi Calls.
